tests/integration/components/language-picker/component-test.js

```diff
@@ -0,0 +1,31 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import Ember from 'ember';
+
+
+const localeStub = Ember.Service.extend({
+  locale: null
+});
+
+moduleForComponent('language-picker', 'Integration | Component | language picker', {
+  integration: true,
+
+  beforeEach: function() {
+    this.register('service:i18n', localeStub);
+    this.inject.service('i18n', { as: 'i18n' });
+  }
+});
+
+test('it sets the locale', function(assert) {
+
+  this.set('selectLanguage', (language, code) => {
+    assert.deepEqual(language, 'English (US)');
+    assert.deepEqual(code, 'en-US');
+  });
+
+  this.render(hbs`{{language-picker onPick=(action selectLanguage)}}`);
```

Comment:
Yes, this is testing the component action, which sets the locale and sends the `selectLanguage` controller action. Lines 22 & 23 assert that the `selectLanguage` action received the correct language name and code.